src/main/java/org/myrobotlab/service/Wikipedia.java

```diff
@@ -188,6 +185,10 @@ public int setMaxImages(int cnt) {
     return cnt;
   }
 
+  @Override
+  public void attach(String serviceName) throws Exception {
+    attachImageListener(serviceName);

```

Comment:
shouldn't this do a type check to see that the service being attached is an image listener?  is that the only thing that wikipedia publishes/ can attach ?  what if it's a search publisher or a text publisher being attached?  what if it's 2 out of 3 of those... or all 3?

Reply:
updated